src/Altinn.Platform/Altinn.Platform.Storage/DataCleanup/Services/CosmosService.cs

```diff
@@ -1,10 +1,125 @@
 using System;
 using System.Collections.Generic;
-using System.Text;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Altinn.Platform.Storage.Interface.Models;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Documents.Linq;
+using Microsoft.Extensions.Logging;
 
 namespace Altinn.Platform.Storage.DataCleanup.Services
 {
-    public static class CosmosService
+    /// <summary>
+    /// Class that handles all interaction with Cosmos DB.
+    /// </summary>
+    public class CosmosService : ICosmosService
     {
+        private readonly string databaseId = "Storage";
+        private readonly string instanceCollectionId = "instances";
+        private readonly string dataElementsCollectionId = "dataElements";
+
+        private readonly DocumentClient _client;
+        private readonly ILogger<ICosmosService> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CosmosService"/> class.
+        /// </summary>
+        /// <param name="logger">The logger</param>
+        public CosmosService(ILogger<ICosmosService> logger)
+        {
+            _logger = logger;
+
+            ConnectionPolicy connectionPolicy = new ConnectionPolicy
+            {
+                ConnectionMode = ConnectionMode.Gateway,
+                ConnectionProtocol = Protocol.Https,
+            };
+            string endpointUri = Environment.GetEnvironmentVariable("EndpointUri");
+            string primaryKey = Environment.GetEnvironmentVariable("PrimaryKey");
+            _client = new DocumentClient(new Uri(endpointUri), primaryKey, connectionPolicy);
+            _client.OpenAsync();
```

Comment:
Don't run async methods in a constructor. In this case, the client might not be ready when it is used.

Reply:
Refactored